src/shape.rs

```diff
@@ -3,16 +3,35 @@
 use crate::{segments, BezPath, Circle, Line, PathEl, Point, Rect, RoundedRect, Segments};
 
 /// A generic trait for open and closed shapes.
+///
+/// This trait provides conversion from shapes to [`BezPath`]s, as well as
+/// general geometry functionality like computing [`area`], [`bounding_box`]es,
+/// and [`winding`] number.
+///
+/// [`BezPath`]: struct.BezPath.html
+/// [`area`]: #tymethod.area
+/// [`bounding_box`]: #tymethod.bounding_box
+/// [`winding`]: #tymethod.winding
 pub trait Shape: Sized {
-    /// The iterator resulting from `path_elements`.
+    /// The iterator returned by the [`path_elements`] method.
+    ///
+    /// [`path_elements`]: #tymethod.path_elements
     type PathElementsIter: Iterator<Item = PathEl>;
 
-    /// Returns an iterator over this shape expressed as Bézier path
-    /// _elements_.
+    /// Returns an iterator over this shape expressed as [`PathEl`]s;
+    /// that is, as Bézier path _elements_.
+    ///
+    /// All shapes can be represented as Béziers, but in many situations
+    /// (such as when interfacing with a platform drawing API) there are more
+    /// efficient native types for specific concrete shapes. In this case,
+    /// the user should exhaust the `as_` methods ([`as_rect`], [`as_line`], etc)
+    /// before converting to a bezier, as those are likely to be more efficient.
```

Comment:
Do we want to be consistent about the spelling (title case and with accent)? I do use "bezier" myself a lot, especially to refer to a concrete representation.